src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -112,13 +112,19 @@ public string GetTypeString()
                     result = $"VarVector<{result}>";
                 return result;
             }
+
+            /// <summary>
+            /// Return if this structure is not identical to the default value of <see cref="Column"/>. If true,
+            /// it means this structure is initialized properly and therefore considered as valid.
+            /// </summary>
+            public bool IsValid => !Equals(default);
```

Comment:
>!Equals(default); [](start = 35, length = 17)

I don't like this usage of `Equals`. This method is on `object`, so and takes an `object`, so there's going to be some pretty wild double boxing going on here. There's a simpler way: in the case where this is not valid, the name will be `null`, so I think we can just use that ... `IsValid => Name != null;`.

Reply:
No problem.

---
In reply to: [238850532](https://github.com/dotnet/machinelearning/pull/1822#discussion_r238850532) [](ancestors = 238850532)